Classes/Flowpack/ElasticSearch/Domain/Model/Index.php

```diff
@@ -18,211 +18,224 @@
-	}
-
-	/**
-	 * @param $typeName
-	 * @return \Flowpack\ElasticSearch\Domain\Model\AbstractType
-	 */
-	public function findType($typeName) {
-		return new GenericType($this, $typeName);
-	}
-
-	/**
-	 * @param array<AbstractType> $types
-	 *
-	 * @return TypeGroup
-	 */
-	public function findTypeGroup(array $types) {
-		return new TypeGroup($this, $types);
-	}
-
-	/**
-	 * @return boolean
-	 */
-	public function exists() {
-		$response = $this->request('HEAD');
-
-		return $response->getStatusCode() === 200;
-	}
-
-	/**
-	 * @return void
-	 */
-	public function create() {
-		$this->request('PUT', NULL, array(), json_encode($this->getSettings()));
-	}
-
-	/**
-	 * @return void
-	 */
-	public function updateSettings() {
-		$settings = $this->getSettings();
-		$updatableSettings = array();
-		foreach ($this->updatableSettings as $settingPath) {
-			$setting = Arrays::getValueByPath($settings, $settingPath);
-			if ($setting !== NULL) {
-				$updatableSettings = Arrays::setValueByPath($updatableSettings, $settingPath, $setting);
-			}
-		}
-		$this->request('PUT', '/_settings', array(), json_encode($updatableSettings));
-	}
-
-	/**
-	 * @return void
-	 */
-	public function delete() {
-		$this->request('DELETE');
-	}
-
-	/**
-	 * Refresh the index
-	 *
-	 * @return void
-	 */
-	public function refresh() {
-		$this->request('POST', '/_refresh');
-	}
-
-	/**
-	 * @param string $method
-	 * @param string $path
-	 * @param array $arguments
-	 * @param string $content
-	 *
-	 * @throws \Flowpack\ElasticSearch\Exception
-	 * @return \Flowpack\ElasticSearch\Transfer\Response
-	 */
-	public function request($method, $path = NULL, $arguments = array(), $content = NULL) {
-		if ($this->client === NULL) {
-			throw new Exception('The client of the index "' . $this->name . '" is not set, hence no requests can be done.');
-		}
-		$path = '/' . $this->name . ($path ?: '');
-
-		return $this->client->request($method, $path, $arguments, $content);
-	}
-
-	/**
-	 * @return string
-	 */
-	public function getName() {
-		return $this->name;
-	}
-
-	/**
-	 * @param Client $client
-	 */
-	public function setClient($client) {
-		$this->client = $client;
-	}
-
-	/**
-	 * @param $settingsKey
-	 */
-	public function setSettingsKey($settingsKey) {
-		$this->settingsKey = $settingsKey;
-	}
-
-	/**
-	 * @return array|null
-	 */
-	protected function getSettings() {
-		if ($this->client instanceof Client) {
-			$settings = Arrays::getValueByPath($this->settings, 'indexes.' . $this->client->getBundle() . '.' . $this->settingsKey) ?: Arrays::getValueByPath($this->settings, 'indexes.default' . '.' . $this->settingsKey);
-		} else {
-			$settings = Arrays::getValueByPath($this->settings, 'indexes.default' . '.' . $this->settingsKey);
-		}
-		return $settings;
-	}
+class Index
+{
+    /**
+     * @var array
+     * @see http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/indices-update-settings.html
+     */
+    protected $updatableSettings= array(
```

Comment:
Good catch, the index was missing before ;) but I fix this force push and merge this one